include/mra/misc/convolutiondata.h

```diff
@@ -0,0 +1,213 @@
+#ifndef MRA_CONVOLUTIONDATA_H
+#define MRA_CONVOLUTIONDATA_H
+
+#include "mra/ops/inner.h"
+#include "mra/misc/gl.h"
+#include "mra/misc/hash.h"
+#include "mra/misc/misc.h"
+#include "mra/misc/types.h"
+#include "mra/misc/adquad.h"
+#include "mra/misc/twoscale.h"
+#include "mra/misc/platform.h"
+#include "mra/misc/autocorr.h"
+#include "mra/tensor/tensorview.h"
+
+#define MRA_MAX_LX 7
+namespace mra {
+
+  template <typename T>
+  struct ConvolutionData {
+    Tensor<T, 2> R, S;
+  };
+
+  template <typename T, Dimension NDIM>
+  class Convolution {
+
+    private:
+      size_type K;
+      Level n;
+      int npt;                                      // number of quadrature points
+      T expnt;
+      T coeff;
+      Translation lx;
+      const T* quad_x;                              // quadrature points
+      const T* quad_w;                              // quadrature weights
+      Tensor<T, 3> autocorrcoef;                    // autocorrelation coefficients
+      Tensor<T, 3> c;                               // autocorrelation coefficients
+      FunctionData<T, NDIM> functiondata;           // function data
```

Comment:
We don't need to create another instance of FunctionData just for the convolution. I suggest to store the reference here and pass the FunctionData to the constructor.